### ISSUE TITLE
Hide comments

### ISSUE THREAD

**Issue by SWW13** [diaspora/diaspora#7553]
Add a feature for poster to hide comments instead of delete.
This could be used in multiple cases such as hiding off-topic comments without deleting them.

Possible legacy implementation:
* Hide action -> comments won't be touched on old pods
* Delete action with hide flag -> comments will be deleted on old pods

Possible additional features:
* Combine multiple hidden messages to "[xxx hidden messages]"
* Add a hide reason such as "This comment was hidden because of <xxx>."

**Comment by spixi** [comment]
I don’t think this is a federation issue, since hiding posts is a user-specific thing. So when I hide a post, it should be only hidden on my pod and only for me. So, this requires a table of user_id, hidable_type and hidable_id. Hidable items include posts and comments.

**Comment by SWW13** [comment]
It's a feature to moderation comments on own posts just like the delete function but less destructive.

Local hide comments for a user it self would be another feature, which indeed does not require federation changes.

**Comment by spixi** [comment]
Ok, thank you @SWW13. Then I misunderstood your FR. But local hide sounds like an interesting feature nontheless.

**Comment by SWW13** [comment]
Feel free to create an issue if there is no existing one.

**Comment by spixi** [comment]
@SWW13 No problem, I found out, that there actually IS a local hide feature, but I have not find out yet how to unhide hidden posts ...

**Comment by spixi** [comment]
Uh, it seems that the „local hide feature“ I found terminates the contact relationship. Sorry @Elektrolli, you were my lab rat and you are not sharing with me anymore  ☹ ...

**Comment by SuperTux88** [comment]
@spixi can you please stop using this issue for off-topic stuff? As @SWW13 already wrote, this issue is about a moderation feature and has nothing to do with a local hide feature. And no, the local hide feature doesn't terminate contact relationships, it only hides the post from your stream, so elektrolli probably just doesn't share with you.
